DICOM/src/Microsoft.Health.DeID.SharedLib/PerturbFunction.cs

```diff
@@ -0,0 +1,106 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using MathNet.Numerics.Distributions;
+using Microsoft.Health.Dicom.DeID.SharedLib.Exceptions;
+using Microsoft.Health.Dicom.DeID.SharedLib.Models;
+
+namespace Microsoft.Health.Dicom.DeID.SharedLib
+{
+    public class PerturbFunction
+    {
+        private readonly PerturbSetting _perturbSetting;
+
+        public PerturbFunction(PerturbSetting perturbSetting = null)
+        {
+            _perturbSetting = perturbSetting ?? new PerturbSetting();
+            _perturbSetting.Validate();
+        }
+
+        public AgeObject Perturb(AgeObject value)
+        {
+            var result = Perturb(value.Value);
+            return new AgeObject(result, value.AgeType);
+        }
+
+        public decimal Perturb(decimal value)
+        {
+            var noise = (decimal)GenerateNoise((double)value);
+            return Math.Round(value + noise, _perturbSetting.RoundTo);
+        }
+
+        public string Perturb(string value)
+        {
+            EnsureArg.IsNotNull(value, nameof(value));
+
+            if (decimal.TryParse(value, out decimal originValue))
+            {
+                var perturbedValue = Perturb(originValue);
+                return perturbedValue.ToString();
+            }
+            else
+            {
+                throw new DeIDFunctionException(DeIDFunctionErrorCode.PerturbFailed, "The value could not be perturbed.");
```

Comment:
🟢 [Discussion]
It's actually the value is not a valid number.
"could not be perturbed" doesn't provide more information than PerturbFailed error code.